src/app/view/perspectives/search/documents/search-documents.component.ts

```diff
@@ -45,28 +46,38 @@ export class SearchDocumentsComponent implements OnInit, OnDestroy {
   @ViewChild('xlTemplate') xlTempl: TemplateRef<any>;
 
   public size: SizeType;
-  public documents: SearchDocument[];
-  private routerSubscription: Subscription;
+  public documents: DocumentModel[];
+  private querySubscription: Subscription;
 
   constructor(private searchService: SearchService,
               private store: Store<AppState>,
               private collectionService: CollectionService,
               private userSettingsService: UserSettingsService) {
+    this.querySubscription = this.store.select(selectQuery)
+      .subscribe(query => {
+        this.store.dispatch(new CollectionsAction.Get({query: query}));
+        this.store.dispatch(new DocumentsAction.Get({query: query}));
+      });
+    Observable.combineLatest(this.store.select(selectAllDocuments), this.store.select(selectAllCollections))
+      .subscribe((values) => {
+        const [documents, collections] = values;
+        if(document){
+          this.documents = documents;
+          if(collections){
+            this.initCollectionsInDocuments(collections);
```

Comment:
Isn't this method breaking the store objects immutability that is one of the basic concepts of NgRx? Wouldn't it be better to map the array of documents to a new array with their copies enriched by collection info?

Reply:
Just to make it clear. In this case it would not make sense to keep the copy of collection information in the documents store since it might become stale. That's why I'm suggesting creating enriched copies of document objects just for UI rendering so the collection information will be refreshed every time there is some change.